src/diagnostics.ts

```diff
@@ -128,6 +122,32 @@ export class Diagnostics {
     await Lifecycle.getInstance().emit('Doctor:diagnostic', { testName, status });
   }
 
+  public async networkCheck(): Promise<void> {
+    await Promise.all(
+      [
+        // salesforce endpoints
+        'https://test.salesforce.com',
+        'https://appexchange.salesforce.com/services/data',
+        // npm and yarn registries
```

Comment:
Definitely agree on using jsforce conn.request for the first 2, since that's what we'd do in the code

I think the last 3 should mimic whatever we're doing in the real code.
ex: plugin-trust is callling npm cli to get the info (and that's going to be influenced by the npm proxy settings, maybe not exactly what jsforce is using).  https://github.com/salesforcecli/plugin-trust/blob/8b583e13dac80995058e14eabf5eff1a1646376e/src/shared/npmCommand.ts#L63-L64

oclif/plugin-update uses `got` and doesn't do any proxy (maybe that's a whole other issue?) so you might replicate that setup?